### DISCUSSION TITLE
Clarifications on snap errors

### DISCUSSION THREAD

**Discussion by superm1** [fwupd/fwupd#6895] [Q&A]
@superm1 Re-opening as I need some clarifications.

The problem with the `1.7.9` version appeared to be fixed be after a restart today it showed up again. So, I though maybe this older shipped version is not the appropriate one for more up-to-date kernels/versions of Ubuntu. Hence, I `apt purge fwupd` then installed the Software version `1.9.14`.

At this point the message disappeared, the nice GUI icon is there, but t the same
![Screenshot from 2024-03-05 20-20-18](https://github.com/fwupd/fwupd/assets/98895614/b0af143e-c29d-4b1f-8839-2be6a69de46b)
appears again...

After a bit of research, I came across several posts talking about `Discover` a package for Ubuntu that should help with apps management I suppose? In this package there is the option in the Settings to _disable vendo-directory_ under Firmware Updates.

This seems to definitively solve the problem. But, for one what is this doing and secondly is it safe to leave as it is now? Please any insight would be great. let me know. Thanks in advance!

_Originally posted by @Overcraft90 in https://github.com/fwupd/fwupd/issues/6874#issuecomment-1983542757_

**Comment by superm1** [comment]
I need to understand what you actually installed and what is showing you that message.  Please provide a screenshot of what you're talking about, not just the message and collect the below:

What does the following show:

```
apt policy fwupd
snap info fwupd
snap info firmware-updater
snap info gnome-firmware
```

**Comment by Overcraft90** [comment]
@superm1 Yeah these are the output messages for the above. In order:

```fwupd:
  Installed: 1.7.9-1~22.04.3
  Candidate: 1.7.9-1~22.04.3
  Version table:
 *** 1.7.9-1~22.04.3 500
        500 http://us.archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages
        100 /var/lib/dpkg/status
     1.7.5-3 500
        500 http://us.archive.ubuntu.com/ubuntu jammy/main amd64 Packages
```
```
name:      fwupd
summary:   Firmware updates for Linux
publisher: Richard Hughes
store-url: https://snapcraft.io/fwupd
contact:   https://github.com/fwupd/fwupd
license:   LGPL-2.1+
description: |
  The full fwupd firmware updating stack for Linux.  Can be used to perform updates on all supported
  devices via https://fwupd.org
commands:
  - fwupd.fwupdmgr
  - fwupd.fwupdtool
services:
  fwupd: dbus, enabled, active
snap-id:      HpOj37PuyuaMUZY0NQhtwnp7oS5P8u5R
tracking:     latest/stable
refresh-date: today at 09:51 CET
channels:
  latest/stable:    1.9.14                2024-03-02 (5749) 23MB -
  latest/candidate: 1.9.14                2024-03-02 (5749) 23MB -
  latest/beta:      ↑                                            
  latest/edge:      1.9.10-504-g67a911b62 2024-03-07 (5784) 24MB -
  1.9.x/stable:     1.9.14                2024-03-02 (5749) 23MB -
  1.9.x/candidate:  1.9.14                2024-02-26 (5749) 23MB -
  1.9.x/beta:       ↑                                            
  1.9.x/edge:       1.9.14-20-g995bb4e36  2024-03-07 (5783) 23MB -
installed:          1.9.14                           (5749) 23MB -
```
```
name:      firmware-updater
summary:   Firmware Updater
publisher: Canonical✓
store-url: https://snapcraft.io/firmware-updater
contact:   https://github.com/canonical/firmware-updater/
license:   GPL-3.0+
description: |
  Update Firmware
snap-id: EI0D1KHjP8XiwMZKqSjuh6W8zvcowUVP
channels:
  latest/stable:    0+git.90251bb 2024-02-20 (121) 11MB -
  latest/candidate: 0+git.90251bb 2024-02-02 (121) 11MB -
  latest/beta:      ↑                                   
  latest/edge:      0+git.019bf66 2023-07-17 (105) 11MB -
```
```
name:      gnome-firmware
summary:   Install firmware on devices
publisher: Mario Limonciello (superm1)
store-url: https://snapcraft.io/gnome-firmware
license:   GPL-2.0+
description: |
  Update, reinstall and downgrade firmware on devices supported by fwupd.
snap-id: moJzcnWtQMfz2dkrQTBQQ6Lg7mjq0ZI2
channels:
  latest/stable:    45.0-16-g1e19cec 2023-10-16  (8) 1MB -
  latest/candidate: ↑                                    
  latest/beta:      ↑                                    
  latest/edge:      45.0-51-g192363f 2024-01-28 (35) 1MB -
```
What is showing me the message is the Software app installed through: `sudo apt install --install-suggests gnome-software`, see image below:
![Screenshot from 2024-03-07 15-29-20](https://github.com/fwupd/fwupd/assets/98895614/348b1a4a-2fbb-488d-883b-c48f3a46ab82)

Also, what I installed is this:
![Screenshot from 2024-03-07 15-30-23](https://github.com/fwupd/fwupd/assets/98895614/f2321162-0d67-42cf-b3c9-1861e9e17a19)

**Reply by superm1** [reply]
OK so you have installed the fwupd snap and that's what gnome-software is using.  Again, pick one or the other; not both.

**Reply by Overcraft90** [reply]
And that's what I don't understand... I only installed one version through the Software app I mentioned never even open the Ubuntu Software in years but somehow the two are sync. After the `apt purge fwupd` I also installed the app from Software.

**Reply by superm1** [reply]
The software app must have installed the snap version, and the apt one is still here.

**Reply by Overcraft90** [reply]
Right, and how can I remove the apt one? If I do a `apt purge fwupd` everything will be removed also the one I installed through the Software app... to be fair now there have been a couple of updates and upon rebooting the system few time it appears the problem is solved. I think for now I leave it as it is, and keep monitoring the situation.

**Comment by Overcraft90** [comment]
What blinks to my eyes is that the first command still shows `v1.7.9`!?

**Reply by superm1** [reply]
As I said before; you need to remove the version from apt if you're using the snap.  Don't install both.

**Comment by Overcraft90** [comment]
As a quick note, I just performed the `snap connect snap-store:fwupd fwupd:fwupd` which I forgot till now. This time after the clean install everything went well. Still, the problem persist. I'm trying to reboot and see.

**Comment by superm1** [comment]
I _think_ that the GUI issue you're mentioning is probably a bug in an older version of gnome-software not understanding newer features in fwupd 1.9.14.  You can disable the vendor-directory remote if you don't need it.

**Reply by Overcraft90** [reply]
I'm not sure what the _vendor-directory remote_ is to be fair?

**Reply by superm1** [reply]
It's a remote supported by newer fwupd versions for the vendor to drop a CAB file to be used. You have no use for it and can disable it using fwupdmgr disable-remote

**Reply by Overcraft90** [reply]
Indeed, I see. Will do and keep the situation under control. Thanks a lot for the assistance for this issue!
